lib/fs/promises.js

```diff
@@ -473,6 +473,7 @@ async function appendFile(path, data, options) {
 
 async function readFile(path, options) {
   options = getOptions(options, { flag: 'r' });
+  options.flag = options.flag || 'r';
```

Comment:
This will actually manipulate the passed in options because `getOptions` does not copy the options object.
Please do:
```js
if (options.flag == null) {
  options = copyObject(options);
  options.flag = 'r';
}
```